astroid/bases.py

```diff
@@ -120,7 +122,11 @@ def infer(self, context=None):
         yield self
 
 
-def _infer_stmts(stmts, context, frame=None):
+def _infer_stmts(
+    stmts: list[InferenceResult],
+    context: InferenceContext | None,
+    frame: nodes.LocalsDictNodeNG | None = None,
```

Comment:
Are you sure it's `LocalsDictNodeNG`? AFAICT `frame` can at minimum also be a `BaseInstance` object. Maybe something else too.

Reply:
You're right. Changing!